addon/compress.h

```diff
@@ -0,0 +1,20 @@
+#include <vector>
+
+#include "compression_worker.h"  // CompressionResult
+#include "zstd.h"
+
+CompressionResult compress(const std::vector<uint8_t> data, size_t compression_level) {
```

Comment:
No implementations of non-inline methods in header files – if you want to split this out (reasonable), then declare the function in this header (or in _a_ header) and define it in a .cc file